operator/controllers/btpoperator_controller.go

```diff
@@ -95,25 +103,236 @@ var (
+
+	newGvksAsStr, err := gvksutils.GvksToStr(newGvks)
+	if err != nil {
+		return err
+	}
+
+	r.setBtpManagerConfigMap(ctx, configMap, newChartVersion, newChartVersion, newGvksAsStr, newGvksAsStr)
+
+	if err := r.Create(ctx, configMap); err != nil {
+		return err
+	}
+
+	r.currentVersion = newChartVersion
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) handleExistingConfigMap(ctx context.Context, configMap *corev1.ConfigMap, newVersion string, newGvks []schema.GroupVersionKind) (bool, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("btp manager config exists")
+
+	currentVersion, ok := configMap.Data[currentCharVersionKey]
+
+	if !ok {
+		return false, fmt.Errorf("%s should be present in configmap but it is not", currentCharVersionKey)
+	}
+
+	currentGvksAsText, ok := configMap.Data[currentGvksKey]
+	if !ok {
+		return false, fmt.Errorf("%s should be present in configmap but it is not", currentGvksAsText)
+	}
+
+	newGvksAsText, err := gvksutils.GvksToStr(newGvks)
+	if err != nil {
+		return false, err
+	}
+
+	if newVersion != currentVersion {
+		logger.Info(fmt.Sprintf("detected version change: {%s} -> {%s}", currentVersion, newVersion))
+
+		r.setBtpManagerConfigMap(ctx, configMap, currentVersion, newVersion, currentGvksAsText, newGvksAsText)
+
+		if err := r.Update(ctx, configMap); err != nil {
+			return false, err
+		}
+
+		logger.Info("update of config map succeed")
+
+		r.currentVersion = newVersion
+
+		return true, nil
+	}
+
+	r.currentVersion = currentVersion
+
+	return false, nil
+}
+
+func (r *BtpOperatorReconciler) setBtpManagerConfigMap(ctx context.Context, configMap *corev1.ConfigMap, oldChartVersion, currentCharVersion, oldGvksStr, currentGvks string) {
+	if configMap == nil {
+		return
+	}
+
+	configMap.Data[oldChartVersionKey] = oldChartVersion
+	configMap.Data[oldGvksKey] = oldGvksStr
+	configMap.Data[currentCharVersionKey] = currentCharVersion
+	configMap.Data[currentGvksKey] = currentGvks
+}
+
+func (r *BtpOperatorReconciler) deleteOrphanedResources(ctx context.Context, configMap *corev1.ConfigMap) error {
+	logger := log.FromContext(ctx)
+	logger.Info("deletion of orphaned resources started")
+
+	oldVersion, ok := configMap.Data[oldChartVersionKey]
+	if !ok {
+		return fmt.Errorf("%s should be present in configmap but it is not", oldChartVersionKey)
+	}
+
+	oldGvksText, ok := configMap.Data[oldGvksKey]
+	if !ok {
+		return fmt.Errorf("%s should be present in configmap but it is not", oldGvksKey)
+	}
+
+	oldGvks, err := gvksutils.StrToGvks(oldGvksText)
+	if err != nil {
+		return err
+	}
+
+	oldVersionLabel := client.MatchingLabels{chartVersionKey: oldVersion}
+
+	numberOfDeletedItems := 0
+	for _, gvk := range oldGvks {
+		list := &unstructured.UnstructuredList{}
+		list.SetGroupVersionKind(gvk)
+
+		if err := r.List(ctx, list, client.InNamespace(ChartNamespace), oldVersionLabel); err != nil {
+			if !k8serrors.IsNotFound(err) && !k8serrors.IsMethodNotSupported(err) && !meta.IsNoMatchError(err) {
```

Comment:
ignoring `MethodNotSupported` error may lead to leftover resources, `UnstructuredList` can be empty because of that error after performing `List` and loop with `Delete` won't delete any leftover resource